ccm_web/client/src/components/InlineTextEdit.tsx

```diff
@@ -0,0 +1,104 @@
+import React, { useRef, useState } from 'react'
+import { makeStyles } from '@material-ui/core/styles'
+import { Button, Grid, TextField } from '@material-ui/core'
+import { Edit as EditIcon } from '@material-ui/icons'
+import { useSnackbar } from 'notistack'
+import { CODE_ENTER, CODE_NUMPAD_ENTER, CODE_ESCAPE } from 'keycode-js'
+
+interface InlineTextEditProps {
```

Comment:
Can this be renamed to `CourseNameTextEditProps`? Same goes with function name. That way it can be understood as it is renaming course

Reply:
It's more generic than course renaming, or at least intended to be.  It's basically just an inline editable text field.

Reply:
I need to make the error message something that gets passed in, rather than having a course name specific message, but I was unsure precisely where that should be handled and didn't do that at this point.

Reply:
Being generic is a good goal, but we probably need to rename some variables (e.g. `const [courseName, setCourseName`]) 😆

Reply:
Ok, In case it is generic I saw some of the variable names as `courseName` so I got bit confused. Of it is generic then course rename text should be removed and use more generic terms

Reply:
> Being generic is a good goal, but we probably need to rename some variables (e.g. `const [courseName, setCourseName]`) 😆

Yes good point. I will fix that as well.

Reply:
Yeah.. I agree it is bit difficult to thing through if the backend is still not wired up yet. The InlineTextEdit should take an error message from the calling function and should not include feature specific text. This can be handled when backend API endpoint and LTI support is added

Reply:
I wasn't sure where it made sense to handle error/success, but I will create properties for the error/success messages for now.

Reply:
This should be better now